src/neptune_api/types.py

```diff
@@ -50,4 +69,27 @@ class Response(Generic[T]):
     parsed: Optional[T]
 
 
-__all__ = ["File", "Response", "FileJsonType", "Unset", "UNSET"]
+@define
+class OAuthToken:
+    _expiration_time: float = field(default=0.0, alias="expiration_time", kw_only=True)
+    access_token: str
+    refresh_token: str
+
+    @classmethod
+    def from_tokens(cls, access: str, refresh: str) -> "OAuthToken":
+        # Decode the JWT to get expiration time
+        try:
+            decoded_token = jwt.decode(access, options=DECODING_OPTIONS)
+            expiration_time = float(decoded_token["exp"])
+        except (jwt.ExpiredSignatureError, jwt.InvalidTokenError, KeyError):
+            raise InvalidApiTokenException("Cannot decode the access token")
+
+        return OAuthToken(access_token=access, refresh_token=refresh, expiration_time=expiration_time)
+
+    @property
+    def seconds_left(self) -> float:
+        return self._expiration_time - time.time()
+
+    @property
+    def is_expired(self) -> bool:
+        return self.seconds_left <= MINIMAL_EXPIRATION_SECONDS
```

Comment:
shouldn't this be `return self.seconds_left <= 0` instead?

Reply:
I'm following the previous behavior :wink: We're refreshing a little bit earlier than expected. I'm not sure about the rationale.

Reply:
rationale is meaning of variable